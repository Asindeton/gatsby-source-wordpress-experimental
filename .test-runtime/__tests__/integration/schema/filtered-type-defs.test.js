import fetchGraphql from "gatsby-source-wordpress-experimental/utils/fetch-graphql"

describe(`[gatsby-source-wordpress-experimental] filtered type definitions`, () => {
  test(`Date field resolver is working`, async () => {
    const result = await fetchGraphql({
      url: `http://localhost:8000/___graphql`,
      query: /* GraphQL */ `
        {
          wpPage(id: { eq: "cG9zdDoy" }) {
            id
            databaseId
            year: date(formatString: "YYYY")
            month: date(formatString: "MMMM")
            dayOfMonth: date(formatString: "DD")
            dayOfWeekNumber: date(formatString: "E")
            dayOfWeekName: date(formatString: "dddd")
            date

            yearGmt: date(formatString: "YYYY")
            monthGmt: date(formatString: "MMMM")
            dayOfMonthGmt: date(formatString: "DD")
            dayOfWeekNumberGmt: date(formatString: "E")
            dayOfWeekNameGmt: date(formatString: "dddd")
            dateGmt: date

            yearModified: modified(formatString: "YYYY")
            monthModified: modified(formatString: "MMMM")
            dayOfMonthModified: modified(formatString: "DD")
            dayOfWeekNumberModified: modified(formatString: "E")
            dayOfWeekNameModified: modified(formatString: "dddd")
            dateModified: modified

            yearModifiedGmt: modifiedGmt(formatString: "YYYY")
            monthModifiedGmt: modifiedGmt(formatString: "MMMM")
            dayOfMonthModifiedGmt: modifiedGmt(formatString: "DD")
            dayOfWeekNumberModifiedGmt: modifiedGmt(formatString: "E")
            dayOfWeekNameModifiedGmt: modifiedGmt(formatString: "dddd")
            dateModifiedGmt: modifiedGmt
          }
        }
      `,
    })

    expect(result).toMatchSnapshot()
  })
})
