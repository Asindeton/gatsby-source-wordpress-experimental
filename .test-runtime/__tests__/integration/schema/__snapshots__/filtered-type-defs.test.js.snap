// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[gatsby-source-wordpress-experimental] filtered type definitions Date field resolver is working 1`] = `
Object {
  "data": Object {
    "wpPage": Object {
      "databaseId": 2,
      "date": "2020-02-25T17:54:21",
      "dayOfMonth": "25",
      "dayOfWeekName": "Tuesday",
      "dayOfWeekNumber": "2",
      "id": "cG9zdDoy",
      "month": "February",
      "year": "2020",
    },
  },
}
`;
